rotkehlchen/tests/external_apis/test_alchemy.py

```diff
@@ -0,0 +1,124 @@
+from typing import TYPE_CHECKING
+
+import pytest
+
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_DAI, A_ETH, A_EUR, A_USD
+from rotkehlchen.constants.prices import ZERO_PRICE
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.externalapis.alchemy import Alchemy
+from rotkehlchen.fval import FVal
+from rotkehlchen.types import (
+    ApiKey,
+    ChainID,
+    ExternalService,
+    ExternalServiceApiCredentials,
+    Price,
+    Timestamp,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.history.price import PriceHistorian
+
+
+@pytest.mark.vcr(match_on=['alchemy_api_matcher'])
+@pytest.mark.parametrize('should_mock_price_queries', [False])
+def test_alchemy_historical_prices(price_historian: 'PriceHistorian', session_alchemy: 'Alchemy'):  # pylint: disable=unused-argument
+    eur = A_EUR.resolve_to_asset_with_oracles()
+    eth = A_ETH.resolve_to_asset_with_oracles()
+    usd = A_USD.resolve_to_asset_with_oracles()
+    dai = A_DAI.resolve_to_asset_with_oracles()
+
+    # First query usd price
+    price = session_alchemy.query_historical_price(
+        from_asset=eth,
+        to_asset=usd,
+        timestamp=Timestamp(1597024800),
+    )
+    assert price == Price(FVal('394.3572783286'))
+
+    # Query EUR price
+    price = session_alchemy.query_historical_price(
+        from_asset=eth,
+        to_asset=eur,
+        timestamp=Timestamp(1597024800),
+    )
+    assert price == Price(FVal('394.3572783286') * FVal('0.850618'))
+
+    # Query an evm token
+    price = session_alchemy.query_historical_price(
+        from_asset=dai,
+        to_asset=usd,
+        timestamp=Timestamp(1597024800),
+    )
+    assert price == Price(FVal('1.018248283'))
+
+
+@pytest.mark.vcr(match_on=['alchemy_api_matcher'])
+@pytest.mark.parametrize('should_mock_price_queries', [False])
+def test_alchemy_current_prices(price_historian: 'PriceHistorian', session_alchemy: 'Alchemy'):  # pylint: disable=unused-argument
+    eth = A_ETH.resolve_to_asset_with_oracles()
+    usd = A_USD.resolve_to_asset_with_oracles()
+    dai = A_DAI.resolve_to_asset_with_oracles()
+
+    price = session_alchemy.query_current_price(
+        from_asset=eth,
+        to_asset=usd,
+    )
+    assert price == Price(FVal('3425.0697715654'))
+
+    price = session_alchemy.query_current_price(
+        from_asset=dai,
+        to_asset=usd,
+    )
+    assert price == Price(FVal('0.9990162355'))
+
+
+@pytest.mark.parametrize('should_mock_price_queries', [False])
+def test_alchemy_api_errors(
```

Comment:
this and the next test seem to be querying the live API. Is this intended? They will be slow and could end up being flaky

Reply:
yeah, i'd have vcr-ed it but vcr does not store 4xx responses

plus, to have an heads-up in case their schema changes.

Reply:
The heads up if the schema changes in error responses is not important.

You can definitely modify vcr through a fixture to record 4xx responses if it does not do it by default. Though we do have cassettes with 404 response for example: https://github.com/rotki/test-caching/blob/08df6cdbbad13ebd63d0af58b7fc335749f7eb94/cassettes/unit/test_calendar_reminders/test_ens_expiry_calendar_reminders%5Bens_data2-base_accounts0-ethereum_accounts0%5D.yaml#L1081